### PR TITLE
feat: set server alert policy email attributes

### LINKED ISSUE #66: feat: set email attributes for SQL threat alerts (SNYK-CC-TF-169/170)
Stems from two related SNYK issues for the `azurerm_mssql_server_security_alert_policy`-resource:

- Missing email recipients for SQL threat alerts ([SNYK-CC-TF-169](https://snyk.io/security-rules/SNYK-CC-TF-169))
  - **Severity:** `Low`
  - **Issue:** Azure SQL Threat detection policy is missing email recipients
  - **Impact:** Alerts may not be delivered to the analysts
  - **Resolve:** Set `email_addresses` attribute

- SQL Advanced Threat alerts are not send to admins ([SNYK-CC-TF-170](https://snyk.io/security-rules/SNYK-CC-TF-170))
  - **Severity:** `Low`
  - **Issue:** SQL Advanced Threat alerts are not send to admins
  - **Impact:** Account administrators might not be informed about SQL advanced threats
  - **Resolve:** Set `email_account_admin` attribute to `true`

### PR DESCRIPTION
**Pull request by helenakallekleiv**

### CODE REVIEW

**Comment by helenakallekleiv** [comment]
Test pass ⭐

File: variables.tf [hunk 1]
```diff
@@ -69,6 +69,18 @@ variable "firewall_rules" {
...
 }
 
+variable "server_security_alert_policy_email_addresses" {
+  description = "A list of custom email addresses to which the alert for this SQL server is sent."
+  type        = list(string)
+  default     = []
+}
+
+variable "server_security_alert_policy_email_account_admins" {
+  description = "Is alerts for this SQL server sent to account adminitrators?"
```
Comment (hknutsen) [inline variables.tf]: Clarify who receives alerts:

```suggestion
  description = "Are alerts for this SQL server sent to subscription owners and administrators?"
```

**Review by hknutsen — CHANGES_REQUESTED** [review]
Looks good 🙂 Added a suggested change to one of the variable descriptions.

File: examples/complete/main.tf [hunk 2]
```diff
@@ -47,6 +47,9 @@ module "sql" {
     }
   }
 
+  server_security_alert_policy_email_account_admins = true
+  server_security_alert_policy_email_addresses      = []
```
Comment (hknutsen) [inline examples/complete/main.tf]:
```suggestion
  security_alert_policy_email_account_admins = true
  security_alert_policy_email_addresses      = []
```

File: main.tf [hunk 3]
```diff
@@ -97,11 +97,12 @@ resource "azurerm_mssql_server_extended_auditing_policy" "this" {
...
-  server_name         = azurerm_mssql_server.this.name
-  state               = "Enabled"
-  disabled_alerts     = []
-  email_addresses     = []
+  resource_group_name  = azurerm_mssql_server.this.resource_group_name
+  server_name          = azurerm_mssql_server.this.name
+  state                = "Enabled"
+  disabled_alerts      = []
+  email_addresses      = var.server_security_alert_policy_email_addresses
+  email_account_admins = var.server_security_alert_policy_email_account_admins
```
Comment (hknutsen) [inline main.tf]:
```suggestion
  email_addresses      = var.security_alert_policy_email_addresses
  email_account_admins = var.security_alert_policy_email_account_admins
```

File: variables.tf [hunk 4]
```diff
@@ -69,6 +69,18 @@ variable "firewall_rules" {
...
+  type        = list(string)
+  default     = []
+}
+
+variable "server_security_alert_policy_email_account_admins" {
+  description = "Are alerts for this SQL server sent to subscription owners and administrators?"
+  type        = bool
+  default     = true
+}
+
```
Comment (hknutsen) [inline variables.tf]:
```suggestion
variable "security_alert_policy_email_addresses" {
  description = "A list of custom email addresses to which the alert for this SQL server is sent."
  type        = list(string)
  default     = []
}

variable "security_alert_policy_email_account_admins" {
  description = "Are alerts for this SQL server sent to subscription owners and administrators?"
  type        = bool
  default     = true
}

```

**Review by hknutsen — CHANGES_REQUESTED** [review]
Could also omit `server_` from variable names, since these variables only apply to the server and thus we do not need to specify it.

**Comment by helenakallekleiv** [comment]
Tests passed 👍🏼

**Review by hknutsen — APPROVED** [review]
Looks good 🎉

**Pull request merged**